tree.go

```diff
@@ -0,0 +1,62 @@
+package rsmt2d
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"github.com/NebulousLabs/merkletree"
+)
+
+var DefaultTreeCreator = &defaultTreeCreator{}
+
+type Tree interface {
+	Push(data []byte)
+	// TODO(ismail): is this general enough?
+	Prove(idx int) (merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64)
+	Root() []byte
+}
```

Comment:
Please don't merge yet. Trying to distill a minimal interface. This one is basically nebolouslabs' API (and these are the methods used).